### ISSUE TITLE
Suggestion: Tag Groups / Bundles

### ISSUE THREAD

**Issue by Andrewskiz** [shaarli/Shaarli#412]
It would be cool if you could group a number of tags into a tag bundle like in Delicious or Pinboard. Also, a way to pin said bundle to the main page of Shaarli so it shows up first.

**Comment by nodiscc** [comment]
@Andrewskiz Would this https://github.com/shaarli/Shaarli/issues/359 work? (setup a custom button linking to a custom tag search)

**Comment by Andrewskiz** [comment]
That would be a useful feature also, however I was thinking along the lines of having the links in the group themselves show up first on the main page.

**Comment by Andrewskiz** [comment]
That would be a useful feature also, however I was thinking along the lines of having the links in the group themselves show up first on the main page.

**Comment by nodiscc** [comment]
sticky/pinned posts -> https://github.com/shaarli/Shaarli/issues/186 is that it?

**Comment by Andrewskiz** [comment]
Yeah, if that would apply to a tag or group of tags that would be it.

**Comment by virtualtam** [comment]
+1 for the feature ;-)

Introducing a tag hierarchy would also be nice when searching/filtering content, e.g.:
```
music
  |- classical
  |- death-guinguette
  |- jazz
      |- be-bop
      |- john-zorn
      |- scat
  |- opera
```

This would require some development though, either:
- maintaining a "tag group" array and hack our way to implement a usable interface;
- introducing a proper relational model for Shaarli entities (related: #351).

**Comment by nodiscc** [comment]
> Introducing a tag hierarchy would also be nice when searching/filtering content

IMHO this would introduce unneeded complexity. If you need to work like this, why not simply rename these tags (`be-bop`,`john-zorn`...) to `jazz/be-bop`, `jazz/john-zorn` (single tag) or `jazz be-bop`,`jazz john-zorn` (convert 1 tag to multiple tags).

Tags in their current form ("single dimension"/flat) are very powerful and let you organize links exactly the way you want. What @Andrewskiz needs is a way to

1. select all links tagged `favorite`,`yeah`,`nice`
 * requires a "batch edit" mode, allowing selection and edition of multiple items. Not easy to implement and needs to be designed carefully. Already reported at https://github.com/shaarli/Shaarli/issues/106 I think.
2. apply the "sticky/pinned" status to each of those. How sticky links work still has to be decided
  * Add a `sticky` property/checkbox to each link (directly in the array, is that even possible?)
  * Always display links that have a preset tag (eg `sticky` or `*`?... make the list customizable) at the top of the link list. This way we can even skip 1. and directly use the `Rename tag` feature.

Anyway I think this use case can be solved by working on https://github.com/shaarli/Shaarli/issues/186 and/org #359 first; nested tags would be really hard to manage.

**Comment by ArthurHoaro** [comment]
I agree with @nodiscc.

You can do almost whatever you want with current tag system and research.
Introducing a tag hierarchy would be complex and need quite some dev (especially the admin part). And at best it would save a few clicks.

About sticky tags, I'm not a big fan of the idea. IMHO the point of the #186 feature is allowing to have one or two shaares on top as "welcome" messages. It really can be done manually without requiring a specific tag pinning feature. And if you really want to set fifty posts as sticky, well, that's up to you and you can.

**Comment by nodiscc** [comment]
> Add a sticky property/checkbox to each link (directly in the array, is that even possible?)

is that even possible?

**Comment by ArthurHoaro** [comment]
Why wouldn't it be?
